network/backend/lightningd/lightningd.go

```diff
@@ -0,0 +1,249 @@
+			wb.ConfirmedBalance += amount
+		} else {
+			wb.UnconfirmedBalance += amount
+		}
+		wb.TotalBalance += amount
+		return true
+	})
+
+	return &wb, nil
+}
+
+func (l Backend) GetChannelsBalance(ctx context.Context) (*models.ChannelsBalance, error) {
+	resp, err := l.client.Call("listfunds")
+	if err != nil {
+		return nil, err
+	}
+
+	balance := models.ChannelsBalance{}
+	resp.Get("channels").ForEach(func(_, ch gjson.Result) bool {
+		balance.Balance += ch.Get("channel_sat").Int()
+		return true
+	})
+
+	return &balance, nil
+}
+
+func (l Backend) ListChannels(ctx context.Context, opt ...options.Channel) ([]*models.Channel, error) {
+	resp, err := l.client.Call("listpeers")
+	if err != nil {
+		return nil, err
+	}
+
+	channels := make([]*models.Channel, 0, resp.Get("peers.#").Int())
+	i := 0
+	resp.Get("peers").ForEach(func(_, peer gjson.Result) bool {
+		if peer.Get("channels.#").Int() == 0 {
+			return true
+		}
+		ch := peer.Get("channels.0")
+
+		channels = append(channels, &models.Channel{
+			ID:                  uint64(i),
+			Status:              models.ChannelActive,
+			RemotePubKey:        peer.Get("id").String(),
+			ChannelPoint:        ch.Get("short_channel_id").String(),
+			Capacity:            ch.Get("msatoshi_total").Int(),
+			LocalBalance:        ch.Get("msatoshi_to_us").Int(),
+			RemoteBalance:       (ch.Get("msatoshi_total").Int() - ch.Get("msatoshi_to_us").Int()),
+			CommitFee:           0,
+			CommitWeight:        0,
+			FeePerKiloWeight:    0,
+			UnsettledBalance:    0,
+			TotalAmountSent:     ch.Get("in_msatoshi_fulfilled").Int(),
+			TotalAmountReceived: ch.Get("out_msatoshi_fulfilled").Int(),
+			ConfirmationHeight:  nil,
+			UpdatesCount:        uint64(ch.Get("in_payments_fulfilled").Int() + ch.Get("out_payments_fulfilled").Int()),
+			CSVDelay:            uint32(ch.Get("our_to_self_delay").Int()),
+			Private:             ch.Get("private").Bool(),
+			PendingHTLC:         []*models.HTLC{},
+			LastUpdate:          nil,
+			Node:                nil,
+			Policy1: &models.RoutingPolicy{
+				MinHtlc: ch.Get("htlc_minimum_msat").Int(),
+			},
+			Policy2: &models.RoutingPolicy{
+				MinHtlc: ch.Get("htlc_minimum_msat").Int(),
+			},
+		})
+
+		i++
+		return true
+	})
+
+	return channels, nil
+}
+
+func (l Backend) GetChannelInfo(ctx context.Context, channel *models.Channel) error {
+	res, err := l.client.Call("listchannels", channel.ChannelPoint)
+	if err != nil {
+		return err
+	}
+
+	// the channel is represented as if it was two
+	fromUs := res.Get(`channels.#[destination=="` + channel.RemotePubKey + `"]`)
+	toUs := res.Get(`channels.#[source=="` + channel.RemotePubKey + `"]`)
+
+	// last update
+	lastUpdate := fromUs.Get("last_update").Int()
+	if toUs.Get("last_update").Int() > lastUpdate {
+		lastUpdate = toUs.Get("last_update").Int()
+	}
+	t := time.Unix(lastUpdate, 0)
+	channel.LastUpdate = &t
+
+	// routing policies
+	channel.Policy1.TimeLockDelta = uint32(fromUs.Get("delay").Int())
```

Comment:
can you create create an instance of &RoutingPolicy for channel.Policy1 in case it is nil
```go
if channel.Policy1 == nil {
  channel.Policy1 = &models.RoutingPolicy{}
}
```